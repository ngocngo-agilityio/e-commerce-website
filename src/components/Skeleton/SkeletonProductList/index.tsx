// Libs
import { Skeleton, Grid, GridItem } from '@chakra-ui/react';

const SkeletonProductList = (): JSX.Element => {
  return (
    <Grid
      templateColumns={{ base: 'repeat(1, 1fr)', md: 'repeat(4, 1fr)' }}
      rowGap={{ base: '20px', md: '48px' }}
      columnGap="20px"
    >
      {Array.from({ length: 8 }, (_, index) => (
        <GridItem key={index}>
          <Skeleton w="full" aspectRatio={{ base: '382/133', md: '295/410' }} />
        </GridItem>
      ))}
    </Grid>
  );
};

export default SkeletonProductList;
