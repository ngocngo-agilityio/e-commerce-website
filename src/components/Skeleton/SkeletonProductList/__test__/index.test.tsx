// Libs
import { render } from '@testing-library/react';

// Components
import SkeletonProductList from '..';

describe('SkeletonProductList component', () => {
  test('should match snapshot for SkeletonProductList', () => {
    const { container } = render(<SkeletonProductList />);

    expect(container).toMatchSnapshot();
  });
});
