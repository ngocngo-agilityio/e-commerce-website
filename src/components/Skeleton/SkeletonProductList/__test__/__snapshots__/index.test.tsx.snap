// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkeletonProductList component should match snapshot for SkeletonProductList 1`] = `
<div>
  <div
    class="css-fc3ct1"
  >
    <div
      class="css-1gvp5h0"
    >
      <div
        class="chakra-skeleton css-12c9fmn"
      />
    </div>
    <div
      class="css-1gvp5h0"
    >
      <div
        class="chakra-skeleton css-12c9fmn"
      />
    </div>
    <div
      class="css-1gvp5h0"
    >
      <div
        class="chakra-skeleton css-12c9fmn"
      />
    </div>
    <div
      class="css-1gvp5h0"
    >
      <div
        class="chakra-skeleton css-12c9fmn"
      />
    </div>
    <div
      class="css-1gvp5h0"
    >
      <div
        class="chakra-skeleton css-12c9fmn"
      />
    </div>
    <div
      class="css-1gvp5h0"
    >
      <div
        class="chakra-skeleton css-12c9fmn"
      />
    </div>
    <div
      class="css-1gvp5h0"
    >
      <div
        class="chakra-skeleton css-12c9fmn"
      />
    </div>
    <div
      class="css-1gvp5h0"
    >
      <div
        class="chakra-skeleton css-12c9fmn"
      />
    </div>
  </div>
</div>
`;
